encoder/encoder.cpp

```diff
@@ -12,12 +12,12 @@
 #include "mjpeg_encoder.hpp"
 #include "null_encoder.hpp"
 
-Encoder *Encoder::Create(VideoOptions const *options)
+Encoder *Encoder::Create(VideoOptions const *options, const StreamInfo &info)
 {
 	if (strcasecmp(options->codec.c_str(), "yuv420") == 0)
 		return new NullEncoder(options);
 	else if (strcasecmp(options->codec.c_str(), "h264") == 0)
-		return new H264Encoder(options);
+		return new H264Encoder(options, info);
 	else if (strcasecmp(options->codec.c_str(), "mjpeg") == 0)
 		return new MjpegEncoder(options);
 	throw std::runtime_error("Unrecognised codec " + options->codec);
```

Comment:
My sense of symmetry feels slightly uncomfortable that not all the encoders want the info parameter, but I think I can survive!  :)